opennms-services/src/main/java/org/opennms/netmgt/collectd/CollectableService.java

```diff
@@ -354,8 +361,13 @@ private void doRun() {
             }
         }
         
+	// If we are doing strict interval, determine how long the collection
+        // has taken, so we can cut that off of the service interval
+	long diff = 0;
+        if (strictInterval)
+            diff = System.currentTimeMillis() - m_lastScheduledCollectionTime;
     	// Reschedule the service
-        m_scheduler.schedule(m_spec.getInterval(), getReadyRunnable());
+        m_scheduler.schedule(m_spec.getInterval() - diff, getReadyRunnable());
```

Comment:
Good point.  I will limit diff to be m_spec.getInterval() at most, so that the difference between them is always >= 0.